src/instruments/src/Common/simVars.tsx

```diff
@@ -0,0 +1,455 @@
+        case "Boolean":
+            return "bool";
+        case "number":
+        case "Number":
+            return "number";
+        case "Degrees":
+        case "degree":
+            return "degree";
+        case "Percent":
+        case "percent":
+            return "percent";
+        case "Feet":
+        case "feet":
+        case "feets":
+        case "Feets":
+            return "feet";
+        case "Knots":
+        case "knots":
+            return "knots";
+        default:
+            return unit;
+    }
+}
+
+type SimVarSetter = <T extends SimVarValue>(oldValue: T) => T;
+
+type RetrieveSimVar = (name: string, unit: UnitName, force?: boolean, global?: boolean) => SimVarValue;
+type UpdateSimVar = (name: string, unit: UnitName, newValueOrSetter: SimVarValue | SimVarSetter, proxy?: string) => void;
+type RegisterSimVar = (name: string, unit: UnitName, maxStaleness: number, global: boolean) => void;
+type UnregisterSimVar = (name: string, unit: UnitName, maxStaleness: number, global: boolean) => void;
+
+const errorCallback = () => {
+    throw Error('useSimVar was called in a React tree with no SimVarProvider');
+};
+const context = React.createContext<{
+    retrieve: RetrieveSimVar,
+    update: UpdateSimVar,
+    register: RegisterSimVar,
+    unregister: UnregisterSimVar,
+}>({
+    retrieve: errorCallback,
+    update: errorCallback,
+    register: errorCallback,
+    unregister: errorCallback,
+});
+const {Provider: InternalProvider}  = context;
+
+type UnitName = string | any; // once typings is next to tsconfig.json, use those units
+type SimVarValue = number | any;
+type SimVarCache = Record<string, {
+    value: SimVarValue,
+    lastUpdatedAgo: number,
+}>;
+
+declare const SimVar; // this can also be replaced once /typings are available
+
+/**
+ * This component provides the basic functionality for the useSimVar hooks.
+ * By keeping the last known SimVar values inside this provider, we're
+ * effectively caching each SimVar, so that there is no additional overhead when
+ * using multiple useSimVar hooks for the same SimVar.
+ * For improved performance, this component will only trigger renders when the
+ * "update" custom event is emitted through an instrument.
+ */
+const SimVarProvider: React.FC = ({ children }) => {
+    const listeners = React.useRef<Record<string, number[]>>({});
+    const [cache, setCache] = useState<SimVarCache>({});
+
+    useUpdate((deltaTime: number) => {
+        const stateUpdates: Record<string, {
+            value?: SimVarValue,
+            lastUpdatedAgo: number,
+        }> = {};
+
+        for (const [key, intervals] of Object.entries(listeners.current)) {
+            // First, let's check if there are any listeners at all
+            if (!intervals.length) {
+                continue;
+            }
+
+            // The refresh time is given by the *smallest* maximum update
+            // interval.
+            const threshold = Math.min(...intervals);
```

Comment:
**suggestion**: A suggestion I make on the basis that this PR is very much about performance: consider only calculating the threshold for a given key while adding or removing an interval for that key. Effectively this means keeping an array of intervals and one properly `thresholdInterval` (or however you want to call it); and then using `thresholdInterval` here.

Reply:
Similar to above, I think for now I'm more worried about messing this up at runtime, I think that's an optimization here we can land later. Your feedback is very much appreciated, I'm sure we'll do another pass for things like this in future.